File: app/src/main/java/com/icerockdev/babenko/repo/ImageRepository.java
package com.icerockdev.babenko.repo;

import com.icerockdev.babenko.model.entities.ImageItem;

import io.reactivex.Single;

/**
 * Created by spurdo on 11/12/2017.
 */

public interface ImageRepository {
    Single<ImageItem[]> requestImages();
}

File: app/src/main/java/com/icerockdev/babenko/repo/ImageRepository.kt
package com.icerockdev.babenko.repo

import com.icerockdev.babenko.model.entities.ImageItem

import io.reactivex.Single

/**
 * Created by spurdo on 11/12/2017.
 */

interface ImageRepository {
    fun requestImages(): Single<Array<ImageItem>>
}
